### PR TITLE
Zijie lint fix for reports component Phase 6

### PR DESCRIPTION
**Pull request by iTvX**
# Description
Lint fix for Reports component, this is phase 6.

## Files changed
All js/jsx file in /src/components/Reports/TasksTable

eslint test:
CleanShot 2023-09-30 at 18 19 16@2x

# How to Test

To test this pull request:

- Check out the current branch
- Run `npm install` and `npm run start:local` to run this PR locally  
- Log in as an admin user
- Check all the functionality of the report is the same as before, and see if any bugs have appeared. (Please record a video)

(Optional)
- Set up ESLint in VSCode/ IntelliJ IDEA, I personally use IntelliJ IDEA
- Delete .eslintignore file in the root: `rm .eslintignore`
- ` npx eslint "src/components/Reports/TasksTable/*.js*" `

- `git restore .`

I have complete confidence that **the code will pass the ESLint tests.** Therefore, I kindly request that you **focus on testing the frontend**. Specifically, please ensure that the **report functionality remains consistent** with how it operated previously. Thank you for your time and effort!

### CODE REVIEW

**Review by luisarevalo21 — CHANGES_REQUESTED** [review]
Hello @iTvX 
I tested your PR, however, running the npx eslint displays an error, also the eslintignore file doesn't exist. see screenshot. 
Screenshot 2023-10-02 at 7 58 32 PM

**Comment by iTvX** [comment]
> Hello @iTvX I tested your PR, however, running the npx eslint displays an error, also the eslintignore file doesn't exist. see screenshot. Screenshot 2023-10-02 at 7 58 32 PM

Hi @luisarevalo21 ,
Because you are not in the root directory of HighestGoodNetworkApp, the instructions I gave you are for a relative path. You should navigate into the HighestGoodNetworkApp folder.

As I mentioned earlier, I have complete confidence that the code will pass the ESLint tests. This PR is primarily aimed at ensuring the frontend behaves as expected.

**Comment by DavidC0126** [comment]
Hi @iTvX, 

I tested the PR and I found when I clicked a project, I will be directed to a blank page. Other two buttons work as described.
Screenshot 2023-10-04 at 12 00 07 AM
Screenshot 2023-10-04 at 12 00 14 AM
Screenshot 2023-10-04 at 12 00 25 AM


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/63c2dfcb-8cc3-4c3d-b4c8-44ebdd961eb1

**Comment by luisarevalo21** [comment]
> > Hello @iTvX I tested your PR, however, running the npx eslint displays an error, also the eslintignore file doesn't exist. see screenshot. Screenshot 2023-10-02 at 7 58 32 PM
> 
> Hi @luisarevalo21 , Because you are not in the root directory of HighestGoodNetworkApp, the instructions I gave you are for a relative path. You should navigate into the HighestGoodNetworkApp folder.
> 
> I suggest you use `pwd` to determine your current location. The .eslintignore file is a default file; if you can't find it, that means you're in **the wrong directory**.
> 
> As I mentioned earlier, I have complete confidence that the code will pass the ESLint tests. This PR is primarily aimed at ensuring the frontend behaves as expected. Thank you for testing this PR.

@iTvX 

I revised, my testing of this PR and can confirm the eslint works when running. 
Screenshot 2023-10-04 at 8 29 03 PM


However, when testing the project, I am to view the reports section and select, "people" and "teams" as normal but when I select projects, and select a project the app crashes. 
See video for reference. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/56371957-804d-45d2-95e8-8b4379b7bc2b

**Comment by iTvX** [comment]
Hi @luisarevalo21 @DavidC0126 ,

I redo the eslint, those issues should be fixed, could you please double check again? Please let me know your result.

**Review by Shubhu0 — DISMISSED** [review]
Hi @iTvX Reviewed the PR and it is working as intended. No issues or errors in the ESLint as well!

image

**Review by savio-henrique — DISMISSED** [review]
Hi @iTvX , I've tested as the video below, and it was working as intended. Also the ESLint linted with no problems!
terminal

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83360232/71a04d3e-03a9-400a-85f5-5fdda36b64ee

**Review by yi-lin-1234 — APPROVED** [review]
Hello, functionality of the TasksTable pages is the same as before. And, the eslint fix worked as expected. Great work!

Screenshot 2023-10-22 at 4 38 04 PM

File: src/components/Reports/TasksTable/TasksTable.jsx [hunk 1]
```diff
@@ -1,16 +1,48 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/TasksTable.jsx:1]: Same comment as [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1356/files#r1384348686)

File: src/components/Reports/TasksTable/TasksTable.jsx [hunk 2]
```diff
@@ -1,16 +1,48 @@
...
+      placeholder="Any user"
+      searchCallback={value => setOneFilter('users', value)}
+      className="tasks-table-filter-item tasks-table-filter-input"
+      value={filters.users}
+    />
+  );
+}
+
+function TasksTable({ WbsTasksID }) {
+  // eslint-disable-next-line camelcase
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/TasksTable.jsx:45]: Same as above.

File: src/components/Reports/TasksTable/TasksTable.jsx [hunk 3]
```diff
@@ -111,6 +132,7 @@ export const TasksTable = ({ WbsTasksID }) => {
       </div>
 
       <TasksDetail
+        /* eslint-disable-next-line camelcase */
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/TasksTable.jsx:135]: Same as above.

File: src/components/Reports/TasksTable/index.js [hunk 4]
```diff
@@ -1 +1,2 @@
-export { TasksTable } from './TasksTable';
+// eslint-disable-next-line import/prefer-default-export
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/index.js:1]: Same as above.

File: src/components/Reports/TasksTable/selectors.js [hunk 5]
```diff
@@ -1,9 +1,12 @@
+// eslint-disable-next-line camelcase
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/selectors.js:1]: Same as above.

File: src/components/Reports/TasksTable/selectors.js [hunk 6]
```diff
@@ -1,9 +1,12 @@
...
 const get_task_by_wbsId = (WbsTasksID, tasks) => {
+  // eslint-disable-next-line camelcase
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/selectors.js:3]: Same as above.

Comment (iTvX) [reply src/components/Reports/TasksTable/selectors.js]: fixed

File: src/components/Reports/TasksTable/selectors.js (hunk 5 above)
Comment (iTvX) [reply src/components/Reports/TasksTable/selectors.js]: fixed

File: src/components/Reports/TasksTable/index.js (hunk 4 above)
Comment (iTvX) [reply src/components/Reports/TasksTable/index.js]: fixed

File: src/components/Reports/TasksTable/TasksTable.jsx (hunk 3 above)
Comment (iTvX) [reply src/components/Reports/TasksTable/TasksTable.jsx]: fixed

File: src/components/Reports/TasksTable/TasksTable.jsx (hunk 2 above)
Comment (iTvX) [reply src/components/Reports/TasksTable/TasksTable.jsx]: fixed

File: src/components/Reports/TasksTable/TasksTable.jsx (hunk 1 above)
Comment (iTvX) [reply src/components/Reports/TasksTable/TasksTable.jsx]: fixed

**Comment by EvianTan** [comment]
Just tried with the latest backend and this branch, got below error.
image
And tried on the incognito window, cannot login from there.
image

File: src/components/Reports/TasksTable/TasksTable.jsx [hunk 7]
```diff
@@ -37,54 +67,42 @@ export const TasksTable = ({ WbsTasksID }) => {
...
         <h4 className="tasks-table-header">Tasks</h4>
       </div>
       <div className="tasks-table-filters-wrapper">
         <div className="tasks-table-filters">
-          <UserOptions tasks={get_tasks} />
-          <FilterOptions filterName={'classification'} width="180px" />
-          <FilterOptions filterName={'priority'} />
-          <FilterOptions filterName={'status'} />
-
+          {/* eslint-disable-next-line camelcase */}
```
Comment (EvianTan) [inline src/components/Reports/TasksTable/TasksTable.jsx:77]: Looks like you already got the camelcase errors fixed, please also delete line 77/80/88/95

**Comment by EvianTan** [comment]
Merged your other PR, so you'd need to resolve the conflicts here as well.

**Comment by EvianTan** [comment]
Same as this [comment](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1356/files#r1391983106)

**Pull request closed without merging**